packages/block-library/src/blocks/slideshow/preview.js

```diff
@@ -104,33 +106,31 @@ const SlideshowPreview = class extends Component {
 
 		styles.slideshow.minHeight = Math.max( attributesHeight, mediaMinHeight, maxAspectRatio ) + 'px';
 
+		const sliderSettings = {
+      customPaging: function(i) {
+        return (
+          <a> {i + 1}</a>
+        );
+      },
+		  infinite: true,
+      arrows: true,
+      dots: true,
+      speed: 0
+    };
+
+    let slide = getGallerySlideData( galleryImages, this.props );
+
+    if ( source === 'post' && !! posts ) {
+      slide = getPostSlideData( posts, this.props );
+    }
+
 		return (
 			<Fragment>
-				{ !! galleryImages.length && <div className={ classes.join( ' ' ) } style={ styles.slideshow }>
-					<div className="novablocks-slideshow__slider">
-						<div className="novablocks-slideshow__slide">
-							{ previewImage && <Fragment>
-								<SlideshowBackground { ...this.props } />
-								<div className="novablocks-slideshow__foreground novablocks-foreground novablocks-u-content-padding novablocks-u-content-align" style={ styles.foreground }>
-									<div
-										className="novablocks-slideshow__inner-container novablocks-u-content-width"
-										style={ styles.content }
-										dangerouslySetInnerHTML={ {
-											__html:
-												( typeof previewImage.title === 'string' && `<h2>${ previewImage.title }</h2>` || '' ) +
-												( typeof previewImage.caption === 'string' && previewImage.caption || '' )
-										} }>
-									</div>
-								</div>
-							</Fragment> }
-						</div>
-					</div>
-					{ galleryImages.length > 1 && <div className="novablocks-slideshow__controls">
-						<div className="novablocks-slideshow__arrow novablocks-slideshow__arrow--prev novablocks-slideshow__arrow--disabled" onClick={ this.props.onPrevArrowClick }></div>
-						<div className="novablocks-slideshow__arrow novablocks-slideshow__arrow--next novablocks-slideshow__arrow--disabled" onClick={ this.props.onNextArrowClick }></div>
-					</div> }
-				</div> }
-				{ ! galleryImages.length && <GalleryPlaceholder { ...this.props } /> }
+          <div className={ classes.join( ' ' ) } style={ styles.slideshow }>
+             <Slider className="novablocks-slideshow__slider" {...sliderSettings}>
```

Comment:
You should start with this. For cleaner code you should split the two use cases into separate components like so.
```js
            <GallerySlider { ...props } />
            <PostsSlider { ...props } />
```